core/src/test/scala/org/apache/spark/FileSuite.scala

```diff
@@ -700,4 +701,38 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
 
     assert(collectRDDAndDeleteFileBeforeCompute(true).isEmpty)
   }
+
+  test("SPARK-25100: Using KryoSerializer and" +
+      "setting registrationRequired true can lead job failed") {
+    val tempDir = Utils.createTempDir()
+    val inputFile = tempDir.getAbsolutePath + "/input"
+    val textFileOutputDir = tempDir.getAbsolutePath + "/out1"
+    val dataSetDir = tempDir.getAbsolutePath + "/out2"
+
+    val writer = new PrintWriter(new File(inputFile))
+
+    for(i <- 1 to 100) {
+      writer.print(i)
+      writer.write('\n')
+    }
+
+    writer.close()
+
+    val conf = new SparkConf(false).setMaster("local").
+      set("spark.kryo.registrationRequired", "true").setAppName("test")
+    conf.set("spark.serializer", classOf[KryoSerializer].getName)
+
+    val jobConf = new JobConf()
+    jobConf.setOutputKeyClass(classOf[IntWritable])
+    jobConf.setOutputValueClass(classOf[IntWritable])
+    jobConf.set("mapred.output.dir", dataSetDir)
+
+    val sc = new SparkContext(conf)
```

Comment:
`sc` should not be a local variable, since FileSuite.sc is cleaned up from LocalSparkContext whereas local variable is not.